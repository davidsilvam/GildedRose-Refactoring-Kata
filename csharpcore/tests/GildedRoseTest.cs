using Xunit;
using System.Collections.Generic;
using Xunit.Abstractions;//added to debug

using ApprovalTests;
using ApprovalTests.Reporters;

using System.Text;

namespace csharpcore
{
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        private readonly ITestOutputHelper _output;

        public GildedRoseTest(ITestOutputHelper output)
        {
            _output = output;
        }

        [Fact]
        public void foo()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            //  _output.WriteLine("Debug");
            Assert.Equal("foo", Items[0].Name);
        }

        [Fact]
        public void NormalItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "+5 Dexterity Vest",
                    SellIn = 10,
                    Quality = 20
                },
                new Item {
                    Name = "Elixir of the Mongoose",
                    SellIn = 5,
                    Quality = 7
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void AgedBrieItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Aged Brie",
                    SellIn = 2,
                    Quality = 0
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void SulfurasItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = 0,
                    Quality = 80
                },
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = -1,
                    Quality = 80
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void BackstagePassesItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 15,
                    Quality = 20
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 10,
                    Quality = 49
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 5,
                    Quality = 49
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void ConjuredItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item {
                   Name = "Conjured Mana Cake",
                   SellIn = 3,
                   Quality = 6
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }
    }
}