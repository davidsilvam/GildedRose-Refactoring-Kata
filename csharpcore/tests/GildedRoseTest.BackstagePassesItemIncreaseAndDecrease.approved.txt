-------- day 0 --------
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
-------- day 1 --------
Backstage passes to a TAFKAL80ETC concert, 14, 21
Backstage passes to a TAFKAL80ETC concert, 9, 50
Backstage passes to a TAFKAL80ETC concert, 4, 50
-------- day 2 --------
Backstage passes to a TAFKAL80ETC concert, 13, 22
Backstage passes to a TAFKAL80ETC concert, 8, 50
Backstage passes to a TAFKAL80ETC concert, 3, 50
-------- day 3 --------
Backstage passes to a TAFKAL80ETC concert, 12, 23
Backstage passes to a TAFKAL80ETC concert, 7, 50
Backstage passes to a TAFKAL80ETC concert, 2, 50
-------- day 4 --------
Backstage passes to a TAFKAL80ETC concert, 11, 24
Backstage passes to a TAFKAL80ETC concert, 6, 50
Backstage passes to a TAFKAL80ETC concert, 1, 50
-------- day 5 --------
Backstage passes to a TAFKAL80ETC concert, 10, 25
Backstage passes to a TAFKAL80ETC concert, 5, 50
Backstage passes to a TAFKAL80ETC concert, 0, 50
-------- day 6 --------
Backstage passes to a TAFKAL80ETC concert, 9, 27
Backstage passes to a TAFKAL80ETC concert, 4, 50
Backstage passes to a TAFKAL80ETC concert, -1, 0
-------- day 7 --------
Backstage passes to a TAFKAL80ETC concert, 8, 29
Backstage passes to a TAFKAL80ETC concert, 3, 50
Backstage passes to a TAFKAL80ETC concert, -2, 0
-------- day 8 --------
Backstage passes to a TAFKAL80ETC concert, 7, 31
Backstage passes to a TAFKAL80ETC concert, 2, 50
Backstage passes to a TAFKAL80ETC concert, -3, 0
-------- day 9 --------
Backstage passes to a TAFKAL80ETC concert, 6, 33
Backstage passes to a TAFKAL80ETC concert, 1, 50
Backstage passes to a TAFKAL80ETC concert, -4, 0
-------- day 10 --------
Backstage passes to a TAFKAL80ETC concert, 5, 35
Backstage passes to a TAFKAL80ETC concert, 0, 50
Backstage passes to a TAFKAL80ETC concert, -5, 0
-------- day 11 --------
Backstage passes to a TAFKAL80ETC concert, 4, 38
Backstage passes to a TAFKAL80ETC concert, -1, 0
Backstage passes to a TAFKAL80ETC concert, -6, 0
-------- day 12 --------
Backstage passes to a TAFKAL80ETC concert, 3, 41
Backstage passes to a TAFKAL80ETC concert, -2, 0
Backstage passes to a TAFKAL80ETC concert, -7, 0
-------- day 13 --------
Backstage passes to a TAFKAL80ETC concert, 2, 44
Backstage passes to a TAFKAL80ETC concert, -3, 0
Backstage passes to a TAFKAL80ETC concert, -8, 0
-------- day 14 --------
Backstage passes to a TAFKAL80ETC concert, 1, 47
Backstage passes to a TAFKAL80ETC concert, -4, 0
Backstage passes to a TAFKAL80ETC concert, -9, 0
-------- day 15 --------
Backstage passes to a TAFKAL80ETC concert, 0, 50
Backstage passes to a TAFKAL80ETC concert, -5, 0
Backstage passes to a TAFKAL80ETC concert, -10, 0
-------- day 16 --------
Backstage passes to a TAFKAL80ETC concert, -1, 0
Backstage passes to a TAFKAL80ETC concert, -6, 0
Backstage passes to a TAFKAL80ETC concert, -11, 0
-------- day 17 --------
Backstage passes to a TAFKAL80ETC concert, -2, 0
Backstage passes to a TAFKAL80ETC concert, -7, 0
Backstage passes to a TAFKAL80ETC concert, -12, 0
-------- day 18 --------
Backstage passes to a TAFKAL80ETC concert, -3, 0
Backstage passes to a TAFKAL80ETC concert, -8, 0
Backstage passes to a TAFKAL80ETC concert, -13, 0
-------- day 19 --------
Backstage passes to a TAFKAL80ETC concert, -4, 0
Backstage passes to a TAFKAL80ETC concert, -9, 0
Backstage passes to a TAFKAL80ETC concert, -14, 0
-------- day 20 --------
Backstage passes to a TAFKAL80ETC concert, -5, 0
Backstage passes to a TAFKAL80ETC concert, -10, 0
Backstage passes to a TAFKAL80ETC concert, -15, 0
-------- day 21 --------
Backstage passes to a TAFKAL80ETC concert, -6, 0
Backstage passes to a TAFKAL80ETC concert, -11, 0
Backstage passes to a TAFKAL80ETC concert, -16, 0
-------- day 22 --------
Backstage passes to a TAFKAL80ETC concert, -7, 0
Backstage passes to a TAFKAL80ETC concert, -12, 0
Backstage passes to a TAFKAL80ETC concert, -17, 0
-------- day 23 --------
Backstage passes to a TAFKAL80ETC concert, -8, 0
Backstage passes to a TAFKAL80ETC concert, -13, 0
Backstage passes to a TAFKAL80ETC concert, -18, 0
-------- day 24 --------
Backstage passes to a TAFKAL80ETC concert, -9, 0
Backstage passes to a TAFKAL80ETC concert, -14, 0
Backstage passes to a TAFKAL80ETC concert, -19, 0
-------- day 25 --------
Backstage passes to a TAFKAL80ETC concert, -10, 0
Backstage passes to a TAFKAL80ETC concert, -15, 0
Backstage passes to a TAFKAL80ETC concert, -20, 0
-------- day 26 --------
Backstage passes to a TAFKAL80ETC concert, -11, 0
Backstage passes to a TAFKAL80ETC concert, -16, 0
Backstage passes to a TAFKAL80ETC concert, -21, 0
-------- day 27 --------
Backstage passes to a TAFKAL80ETC concert, -12, 0
Backstage passes to a TAFKAL80ETC concert, -17, 0
Backstage passes to a TAFKAL80ETC concert, -22, 0
-------- day 28 --------
Backstage passes to a TAFKAL80ETC concert, -13, 0
Backstage passes to a TAFKAL80ETC concert, -18, 0
Backstage passes to a TAFKAL80ETC concert, -23, 0
-------- day 29 --------
Backstage passes to a TAFKAL80ETC concert, -14, 0
Backstage passes to a TAFKAL80ETC concert, -19, 0
Backstage passes to a TAFKAL80ETC concert, -24, 0
-------- day 30 --------
Backstage passes to a TAFKAL80ETC concert, -15, 0
Backstage passes to a TAFKAL80ETC concert, -20, 0
Backstage passes to a TAFKAL80ETC concert, -25, 0
